assets/src/data/ee-lists/resolvers.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import apiRequest from '@wordpress/api-request';
+
+/**
+ * Internal dependencies
+ */
+import { setRequested, receiveResponse } from './actions';
+import { applyQueryString } from '../model/endpoints';
+
+/**
+ * Resolver for generic items returned from an endpoint.
+ *
+ * @param {Object} state  Data in state.
+ * @param {string} modelName  The name of the model the items are for.
+ * @param {string} queryString  Additional query string parameters passed on to
+ *   the REST request.
+ */
+export async function * getItems( state, modelName, queryString ) {
+	if ( state[ modelName ] &&
+		state[ modelName ].hasOwnProperty( queryString ) &&
+		state[ modelName ][ queryString ] !== null
+	) {
+		yield receiveResponse( modelName,
+			queryString,
+			state[ modelName ][ queryString ],
+		);
+	}
+	yield setRequested( modelName, queryString );
+	const items = await apiRequest( {
+		path: applyQueryString( modelName,
+			queryString,
+		),
+	} );
+	yield receiveResponse( modelName, queryString, items );
+}
+
+/**
+ * Resolver for event entities.
+ *
+ * @param {Object} state Data in state.
+ * @param {string} queryString Additional query string parameters passed on to
+ *   the REST request.
+ * @return {IterableIterator<*>} A async iterable.
+ */
+export function getEvents( state, queryString ) {
+	return getItems( state, 'event', queryString );
+}
```

Comment:
> should this last method go in the same file with the generic getItems() method?

Well it IS in the same file as the `getItems` async genrator and that's what it is calling :).  No the resolvers have to have the same function names as the selectors and they get "automagically" assembled together as a part of the store object when registering them with `wp.data.registerStore`.  In other words, a resolver that matches a selector will query the api ONCE to populate the store on the first call to that selector.  Then further calls to the selectors will just retrieve data from the store.

> and are we going to populate this same file with similar methods for other entity types? 

possibly.  I haven't decided yet.  I want to start using this in actual implementations before I decide on an appropriate pattern.  The tricky part here is that as I pointed out above there must be a matching resolver for any selector that needs populated via the REST api on the initial query.

Another spoke to this (which isn't found in this branch yet) is adding actions are _dispatched_ to the store for updating data.  I don't want to start adding those actions until we start seeing how this will work in full implementations.  

So I've focused on having the generic functions and a single concrete (event) to begin using and then we'll add more as we need them.  Once we start seeing patterns emerge we can work on implementing more DRY architecture.

Reply:
> `should this last method go in the same file with the generic getItems() method?`
> 
> Well it IS in the same file as the getItems async genrator and that's what it is calling :).

sigh... I realize that there are a **lot** of aspects to this new JS code that I don't understand and I am more than willing to admit when there is something I'm not grokking... but I **am** however fully capable of recognizing when two functions are in the same file, **especially** because I almost always checkout the branch and look at the actual code when dealing with a complicated diff or code that is more challenging. In this case it's also obvious that those two functions are in the same file because if they were not, then one of them would have to be called via an imported object (ie: `object.getItems()`) right? 

I think it would benefit you greatly in the future to give people the benefit of the doubt, and take a few seconds every now and then to consider if **maybe** the person you're about to respond to actually **does** understand the situation, and if **maybe** there is some other way that you could interpret what they have written in a way that doesn't presume them to be completely void of intelligence. 

I fully understood that those two functions were in the same file because I was looking at the actual code in my IDE. So what I was actually asking was **SHOULD** that last method go in the same file with the generic getItems() method... as in... does it belong there or should it go in its own file? In other words, I wanted to know if the plan was to have **ALL** of the resolvers in one file, or to separate specific entity types into their own files, ie: a generic parent `resolvers.js` plus `event-resolvers.js`, `datetime-resolvers.js`, `attendee-resolvers.js`, `registration-resolvers.js`, `transaction-resolvers.js`, etc, etc, etc.

The last sentence you wrote above is sufficient in answering my question and there is no need to respond to this.

Reply:
> I think it would benefit you greatly in the future to give people the benefit of the doubt, and take a few seconds every now and then to consider if maybe the person you're about to respond to actually does understand the situation, and if maybe there is some other way that you could interpret what they have written in a way that doesn't presume them to be completely void of intelligence.

And maybe it would benefit you greatly in the future if you don't immediately jump to the conclusion that someone is criticizing your intelligence.  My statement was intended clarify what thought you meant by "same file" since in one paragraph it seemed to refer to different things. I'm left asking "What "same file are you referring to'?",   "Is he asking if this should be in the same file as `selectors.js` (which also has a `getEvents` method already)?".   In the light of vagueness, I made a statement that was a segue into how the methods functioned which hopefully would have clarified why they were named as is.  I honestly did not know _what_ you were referring to or the context of your question so thought my reply might help expedite the conversation and clarify what's happening in these files.  It definitely was _not_ intended in the way you seem to have taken it by your reply.

I hope there could be a reconsideration of the default way my comments seem to be perceived.  My goal is to help advance the project not put others down.